server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -382,16 +384,37 @@ synchronized void stopWatcher() {
         }
     }
 
-    private WatchKey enableSettingsWatcher(WatchKey previousKey, Path settingsDir) throws IOException {
+    // package private for testing
+    long retryDelayMillis(int failedCount) {
+        assert failedCount < 31;
```

Comment:
Worth adding a comment that this is just to protect against overflow, normally `failedCount` should be max 5